pkg/operator/observe_config.go

```diff
@@ -157,20 +197,26 @@ func observeClusterConfig(kubeClient kubernetes.Interface, clientConfig *rest.Co
 	//     networking:
 	//       podCIDR: 10.2.0.0/16
 	//       serviceCIDR: 10.3.0.0/16
-	restrictedCIDRs := []string{}
-	networking, ok := installConfig["networking"].(map[string]interface{})
-	if !ok {
-		return observedConfig, nil
+	restrictedCIDRs, found, err := unstructured.NestedStringSlice(observedConfig, "admissionPluginConfig", "openshift.io/RestrictedEndpointsAdmission", "configuration", "restrictedCIDRs")
+	if err != nil || !found {
+		restrictedCIDRs = []string{"", ""}
 	}
-	if cidr := networking["podCIDR"]; cidr != nil {
-		restrictedCIDRs = append(restrictedCIDRs, fmt.Sprintf("%v", cidr))
+	errs := []error{}
+	podCIDR, _, err := unstructured.NestedString(installConfig, "networking", "podCIDR")
+	if err != nil {
```

Comment:
when you get to two conditions (not just if/else, but if/else if), use a switch case.

Reply:
Done